packages/picasso/src/TreeView/useNodes.ts

```diff
@@ -63,13 +63,21 @@ export const useNodes = (
   rootNode: HierarchyPointNode<TreeNodeInterface>
 ): DynamicPointNode[] => {
   const [initialized, setInitializedState] = useState<boolean>(false)
-  const initialNodes = useMemo(() => {
-    return getDynamicNodes(rootNode.descendants())
-  }, [rootNode])
+  const initialNodes = useRef<DynamicPointNode[] | null>(null)
+
+  // we only need to prepare initial nodes once on a first render
+  if (initialNodes.current === null) {
```

Comment:
How about this for consistency with the next lines?
```suggestion
  if (!initialNodes.current) {
```